### ISSUE TITLE
Tortoise orm assign class Meta IncompatibleVariableOverride from 1.1.339

### ISSUE THREAD

**Issue by mrparalon** [microsoft/pyright#6814]
**Describe the bug**
I'm defining Meta class for tortoise orm, and it gives me an error. Starts from pyright 1.1.339

```
error: "Meta" overrides symbol of same name in class "Model"
    "types_meta.User.Meta" is incompatible with "tortoise.models.Model.Meta"
    Type "type[types_meta.User.Meta]" cannot be assigned to type "type[tortoise.models.Model.Meta]" (reportIncompatibleVariableOverride)
```

**Code or Screenshots**

`pip install pyright==1.1.339 tortoise-orm`

```sh
❯ pip freeze
aiosqlite==0.17.0
iso8601==1.1.0
nodeenv==1.8.0
pypika-tortoise==0.1.6
pyright==1.1.339
pytz==2023.3.post1
tortoise-orm==0.20.0
typing_extensions==4.9.0
```

```python
from tortoise import fields
from tortoise.models import Model


class User(Model):
    user_id = fields.UUIDField()

    class Meta:
        table = "users"
```

**VS Code extension or command-line**
Command line pyright 1.1.339

**Comment by erictraut** [comment]
This behavior is correct, not a bug. You're overriding the symbol `Meta` with another symbol of the same name but a different type. This is a type violation.

This is covered under the `reportIncompatibleVariableOverride` check, which was previously disabled by default but was recently enabled by default to conform to the Python static typing specification. If you would like to disable this check, you can do so by adding `"reportIncompatibleVariableOverride": "none"` to your `pyrightconfig.json` file. For more details about pyright configuration, refer to [this documentation](https://microsoft.github.io/pyright/#/configuration).

**Comment by zmievsa** [comment]
@erictraut I have updated to the latest pylance (v2024.1.104 (pre-release)) and this error does not appear there. In fact, the error sets are significantly different between pylance and pyright in the latest versions.

**Comment by zmievsa** [comment]
@erictraut This option is "error" by default in pyright and false in pylance. Is this expected? I see a similar problem with "reportFunctionMemberAccess".

**Comment by erictraut** [comment]
The `reportIncompatibleVariableOverride` is error by default if you set the type checking mode to "standard" but off by default if the type checking mode is "basic" or "off". Here is a [handy chart](https://microsoft.github.io/pyright/#/configuration?id=diagnostic-rule-defaults) that shows the default severity levels for each type checking mode.

In Pylance, the type checking mode is set to "off" by default, so you shouldn't see any type-related diagnostics if you're using Pylance defaults. I'm guessing that you've manually set the Pylance default to "basic" at some point.

Pyright defaults to "standard" type checking mode by default. 

If you want to get the same defaults in pylance and pyright, you'll need to specify the same type checking mode for both.

**Comment by zmievsa** [comment]
Thanks, @erictraut ! I have used pyright + pylance for quite a while now so I missed the point when the standard option was added :)

This fixes my issue.
